cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/model/Task.java

```diff
@@ -33,6 +35,7 @@ public Task(String name, Map<String, Serializable> preTaskActionData, Map<String
         this.inputs = inputs;
         this.navigationStrings = navigationStrings;
         this.refId = refId;
+        this.async = preTaskActionData.containsKey(ScoreLangConstants.ASYNC_LOOP_KEY);
```

Comment:
what happens if it is not in the map?
the boolean is false?
in general i prefer no logic, and for sure not BL in constructors.